Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup_extension.h

```diff
@@ -0,0 +1,194 @@
+ * @sa `orient_polygon_soup()`
+ */
+template <class PointRange, class PolygonRange>
+bool
+duplicate_incompatible_edges_in_polygon_soup(PointRange& points,
+                                            PolygonRange& polygons)
+{
+  std::size_t inital_nb_pts = points.size();
+  typedef CGAL::Polygon_mesh_processing::internal::
+    Polygon_soup_orienter<PointRange, PolygonRange> Orienter;
+
+  Orienter orienter(points, polygons);
+  orienter.fill_edge_map();
+  // make edges to duplicate
+  for(std::size_t i1=0;i1<points.size();++i1)
+    for(const typename Orienter::Internal_map_type::value_type& i2_and_pids : orienter.edges[i1])
+      if (i2_and_pids.second.size() > 1)
+        orienter.set_edge_marked(i1,i2_and_pids.first,orienter.marked_edges);
+  orienter.duplicate_singular_vertices();
+
+  return inital_nb_pts==points.size();
+}
+
+/*!
+ * \ingroup PMP_orientation_grp
+ * Orients each triangle of a triangle soup using the orientation of its
+ * closest non degenerate triangle in `tm_ref`.
+ * \tparam Concurrency_tag enables sequential versus parallel orientation.
+                        Possible values are `Sequential_tag` (the default) and
+                        `Parallel_tag`.
+ * \tparam PointRange a model of the concepts `RandomAccessContainer`
+ * and `BackInsertionSequence` whose value type is the point type.
+ * @tparam TriangleRange a model of the concept `RandomAccessContainer`
+ * whose value_type is a model of the concept `RandomAccessContainer`
+ * whose value_type is `std::size_t`and of size 3.
+ * @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph` .
+ *
+ * \param tm_ref the reference TriangleMesh.
+ * \param points the points of the soup.
+ * \param triangles the triangles of the soup.
+ * @param np optional sequence of \ref pmp_namedparameters among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{vertex_point_map}
+ *     the property map with the points associated to the vertices of `tm`.
+ *     If this parameter is omitted, an internal property map for
+ *     `CGAL::vertex_point_t` must be available in `PolygonMesh`
+ *   \cgalParamEnd
+ *   \cgalParamBegin{geom_traits} a geometric traits class instance.
+ *      The traits class must provide the nested functor `Collinear_3`
+ *      to check whether three points are collinear.
+ *   \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+
+template <class Concurrency_tag, class PointRange, class TriangleRange,
+          class TriangleMesh, class NamedParameters>
+void
+orient_triangle_soup_with_reference_triangle_mesh(
+  const TriangleMesh& tm_ref,
+  PointRange& points,
+  TriangleRange& triangles,
+    const NamedParameters& np)
+{
+  namespace PMP = CGAL::Polygon_mesh_processing;
+
+  typedef boost::graph_traits<TriangleMesh> GrT;
+  typedef typename GrT::face_descriptor face_descriptor;
+  typedef typename PointRange::value_type Point_3;
+  typedef typename Kernel_traits<Point_3>::Kernel K;
```

Comment:
isn't it possible to use the `geom_traits` given as a named parameter, instead of this `K`?

Reply:
The geom_traits is only needed to provide Collinear_3. If we want to use it, we will have to specify that it musts define the dsame `Point_3` than the vpm, which complexifies the reqs and leads to the exact same type :/

Reply:
Ok. Then you should extract the point type from the vpm, and not using `Kernel_traits`

Reply:
Well actually all points type must be unified. I'll just add it to the doc and use the geom_traits.

Reply:
agreed!